cmd/certficates/main.go

```diff
@@ -0,0 +1,341 @@
+}
+
+func cmdSetup(cmd *cobra.Command, args []string) error {
+	setupDir, err := filepath.Abs(defaultConfDir)
+	if err != nil {
+		return err
+	}
+
+	valid, err := fpath.IsValidSetupDir(setupDir)
+	if err != nil {
+		return err
+	}
+	if !setupCfg.Overwrite && !valid {
+		fmt.Printf("certificate signer configuration already exists (%v). rerun with --overwrite\n", setupDir)
+		return nil
+	}
+
+	err = os.MkdirAll(setupDir, 0700)
+	if err != nil {
+		return err
+	}
+
+	setupCfg.CA.CertPath = filepath.Join(setupDir, "ca.cert")
+	setupCfg.CA.KeyPath = filepath.Join(setupDir, "ca.key")
+	setupCfg.Identity.CertPath = filepath.Join(setupDir, "identity.cert")
+	setupCfg.Identity.KeyPath = filepath.Join(setupDir, "identity.key")
+
+	err = provider.SetupCA(process.Ctx(cmd), setupCfg.CA)
+	if err != nil {
+		return err
+	}
+
+	o := map[string]interface{}{
+		"ca.cert-path":       setupCfg.CA.CertPath,
+		"ca.key-path":        setupCfg.CA.KeyPath,
+		"identity.cert-path": setupCfg.Identity.CertPath,
+		"identity.key-path":  setupCfg.Identity.KeyPath,
+	}
+	return process.SaveConfig(runCmd.Flags(),
+		filepath.Join(setupDir, "config.yaml"), o)
+}
+
+func cmdRun(cmd *cobra.Command, args []string) error {
+	ctx := process.Ctx(cmd)
+
+	return runCfg.Identity.Run(ctx, nil, runCfg.CertSigner)
+}
+
+func cmdCreateAuth(cmd *cobra.Command, args []string) error {
+	count, err := strconv.Atoi(args[0])
+	if err != nil {
+		return errs.New("Count couldn't be parsed: %s", args[0])
+	}
+	authDB, err := authCreateCfg.NewAuthDB()
+	if err != nil {
+		return err
+	}
+
+	var emails []string
+	if len(args) > 1 {
+		if authCreateCfg.EmailsPath != "" {
+			return errs.New("Either use `--emails-path` or positional args, not both.")
+		}
+		emails = args[1:]
+	} else {
+		list, err := ioutil.ReadFile(authCreateCfg.EmailsPath)
+		if err != nil {
+			return errs.Wrap(err)
+		}
+		emails = strings.Split(string(list), authCreateCfg.Delimiter)
+	}
+
+	var incErrs utils.ErrorGroup
+	for _, email := range emails {
+		if _, err := authDB.Create(email, count); err != nil {
+			incErrs.Add(err)
+		}
+	}
+	return incErrs.Finish()
+}
+
+func cmdInfoAuth(cmd *cobra.Command, args []string) error {
+	authDB, err := authInfoCfg.NewAuthDB()
+	if err != nil {
+		return err
+	}
+
+	var emails []string
+	if len(args) > 0 {
+		if authInfoCfg.EmailsPath != "" {
+			return errs.New("Either use `--emails-path` or positional args, not both.")
+		}
+		emails = args
+	} else if _, err := os.Stat(authInfoCfg.EmailsPath); err != nil {
+		return errs.New("Emails path error: %s", err)
+	} else {
+		list, err := ioutil.ReadFile(authInfoCfg.EmailsPath)
+		if err != nil {
+			return errs.Wrap(err)
+		}
+		emails = strings.Split(string(list), authInfoCfg.Delimiter)
+	}
+
+	var emailErrs, printErrs utils.ErrorGroup
+	w := tabwriter.NewWriter(os.Stdout, 0, 2, 2, ' ', 0)
+	if _, err := fmt.Fprintln(w, "Email\tClaimed\tAvail.\t"); err != nil {
+		return err
+	}
+
+	for _, email := range emails {
```

Comment:
maybe pull this out into its own function ? this is a really large function right now